natural_language_processing/word_embeddings/word_embeddings.py

```diff
@@ -0,0 +1,158 @@
+import re
+import glob
+import os
+from gensim.models import word2vec  # word2vec algorithm
+import nltk
+from nltk.corpus import stopwords  # list of words that we exclude from the corpus
+from nltk.tokenize import word_tokenize  # document tokenization
+from sklearn.manifold import TSNE  # for visualizing high-dimensional data
+import matplotlib.pyplot as plt  # for plotting
+import pandas as pd
+nltk.download('stopwords')
+stop_words = set(stopwords.words('english'))
+
+"""
+This is the class of Word Embeddings.
+Model is trained with word2vec algorithm (https://youtu.be/c3yRH0XZN2g)
+"""
+
+
+class WordVectors(object):
+    def __init__(self, model: word2vec.Word2Vec):
+        """
+        Init function which takes in a trained model
+        """
+        self._model = model
+        self._wv = model.wv
+        self._vocab = model.wv.vocab
+
+    def analogy(self, x1: str, x2: str, y1: str) -> str:
+        """
+        Tries to find a word which has a similar relation to y1, as x2 has to x1
+        """
+        x1, x2, y1 = x1.lower(), x2.lower(), y1.lower()
+        error_msg = 'every word must be in the vocabulary'
+        assert all(self.is_in_vocab(x) for x in (x1, x2, y1)), error_msg
+        result = self._wv.most_similar(positive=[y1, x2], negative=[x1])[0][0]
+        return result
+
+    def n_similarity(self, list1: list, list2: list) -> float:
+        """
+        Returns similarity score between two lists of words
+        """
+        list1 = [word.lower() for word in list1 if self.is_in_vocab(word)]
+        list2 = [word.lower() for word in list2 if self.is_in_vocab(word)]
+        error_msg = 'at least one word from each list must be in the vocabulary'
+        assert list1 and list2, error_msg
+        score = self._wv.n_similarity(list1, list2)
+        return score
+
+    def similarity(self, w1: str, w2: str) -> float:
+        """
+        Returns similarity score between two words
+        """
+        w1, w2 = w1.lower(), w2.lower()
+        error_msg = 'both words must be in the vocabulary'
+        assert self.is_in_vocab(w1) and self.is_in_vocab(w2), error_msg
+        score = self._wv.similarity(w1, w2)
+        return score
+
+    def closest_words(self, word: str) -> list:
+        """
+        Returns similar words for a given word
+        """
+        word = word.lower()
+        error_msg = 'the word must be in the vocabulary'
+        assert self.is_in_vocab(word), error_msg
+        close_words = self._wv.similar_by_word(word)
+        return close_words
```

Comment:
```suggestion
        return self._wv.similar_by_word(word)
```
The method name already tells us that we are returning the closest words.